[PT2] - Allowlist should have precedence

flaviotruzzi opened the pull request:
Summary: When working on List[List[int]], the ints were being considered Constants regardless of their inclusion on the allowlist.

Test Plan:
CI + new test

https://www.internalfb.com/intern/testinfra/testrun/5066549856504774

Differential Revision: D73137631




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

flaviotruzzi commented:
@pytorchbot label "topic: not user facing"

laithsakka commented on test/dynamo/test_misc.py:
```diff
@@ -7785,6 +7786,32 @@ def forward(self, x, y):
+        builder._DYNAMIC_SOURCES = None
+
+        counter = CompileCounter()
+
```
use     @torch.compiler.config.patch(dynamic_sources=...) otherwise those will be set forever in the test

flaviotruzzi replied:
Added, still have the _DYNAMIC_SOURCES set to None to clear the cache.

laithsakka reviewed (approved):
just use     @torch.compiler.config.patch() before landing

flaviotruzzi committed:
[PT2] - Allowlist should have precedence (#151942)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/151942

When working on List[List[int]], the ints were being considered Constants regardless of their inclusion on the allowlist.

Test Plan:
CI + new test

https://www.internalfb.com/intern/testinfra/testrun/16044073752153877

Reviewed By: laithsakka

Differential Revision: D73137631

flaviotruzzi commented:
@pytorchbot merge
